src/components/table/template.test.js

```diff
@@ -0,0 +1,362 @@
+/* globals describe, it, expect */
+
+const { render, getExamples } = require('../../../lib/jest-helpers')
+
+const examples = getExamples('table')
+describe('Table', () => {
+  describe('by default', () => {
+    const { $ } = render('table', examples.default)
+    const $component = $('.govuk-c-table')
+    const $tableBody = $component.find('.govuk-c-table__body')
+    const $tableRow = $component.find('.govuk-c-table__row')
+    const $tableCell = $component.find('.govuk-c-table__cell')
+
+    it('renders one tbody element', () => {
+      expect($tableBody).toHaveLength(1)
+    })
+
+    it('renders the specified number of rows', () => {
+      expect($tableRow).toHaveLength(3)
+    })
+
+    it('renders the specified number of cells in row 1', () => {
+      expect($tableRow.eq(0).find($tableCell)).toHaveLength(3)
+    })
+
+    it('renders the specified number of cells in row 2', () => {
+      expect($tableRow.eq(1).find($tableCell)).toHaveLength(3)
+    })
+
+    it('renders the specified number of cells in row 3', () => {
+      expect($tableRow.eq(2).find($tableCell)).toHaveLength(3)
+    })
+
+    it('renders correct text in cell 1 of row 1', () => {
+      expect($tableRow.eq(0).find($tableCell).eq(0).text()).toEqual('January')
+    })
+
+    it('renders correct text in cell 2 of row 1', () => {
+      expect($tableRow.eq(0).find($tableCell).eq(1).text()).toEqual('£85')
+    })
+
+    it('renders correct text in cell 3 of row 1', () => {
+      expect($tableRow.eq(0).find($tableCell).eq(2).text()).toEqual('£95')
+    })
+  })
+
+  describe('variant', () => {
```

Comment:
could we get these two tests outside 'variant', even if they're using examples from yaml. also, and I know I'm being picky with this one, but can we order them somehow in a similar way we have them in the table or arguments (classes, caption..) would make it easier to follow and make sure we cover all the arguments.